src/librbd/ExclusiveLock.cc

```diff
@@ -101,6 +103,10 @@ int ExclusiveLock<I>::get_unlocked_op_error() const {
 template <typename I>
 void ExclusiveLock<I>::init(uint64_t features, Context *on_init) {
   ceph_assert(m_image_ctx.owner_lock.is_locked());
+
+  using klass = ExclusiveLock<I>;
+  on_init = create_ref_counted_context<klass>(this, on_init);
```

Comment:
I do not quite understand this change, is this for avoiding exlusiveLock to be deleted before on_init is called? If so, in what situation will this happen?

Reply:
What is the race condition? Could you give an example ?